source/agora/consensus/Ledger.d

```diff
@@ -766,10 +756,20 @@ public class Ledger
             return "Internal error: Could not calculate random seed at current height";
         }
 
-        auto expected_cb_txs = this.getCoinbaseTX(block.txs, block.header.missing_validators);
         auto incoming_cb_txs = block.txs.filter!(tx => tx.isCoinbase);
-        if (!isPermutation(expected_cb_txs, incoming_cb_txs))
-            return "Invalid Coinbase transaction";
+        // If it is a payout block then a single Coinbase transaction is included
+        if (block.header.height % this.params.PayoutPeriod == 0)
+        {
+            if (incoming_cb_txs.count == 0)
+                return "Missing expected Coinbase transaction in payout block";
+            if (incoming_cb_txs.count > 1)
+                return "There should only be one Coinbase transaction in payout block";
```

Comment:
You should add a test for the second condition, because I think it can never trigger.
`count` may consume the range, especially given that the range has no length (`filter`ed ranges cannot have a `length` property, obviously). So calling `count` twice will either evaluate the range twice (not great, but correct) or once and leave it empty, which means that multiple coinbase txs wouldn't be caught.

Reply:
Suggestion:
```D
const cbTxCount = incoming_cb_txs.count;
if (block.header.height % this.params.PayoutPeriod == 0)
{
    if (cbTxCount != 1)
        return cbTxCount ? "Only one" : "Missing";
}
else if (cbTxCount)
    return "Nope";
```

Reply:
Updated